app/controllers/posts_controller.rb

```diff
@@ -1,19 +1,33 @@
 class PostsController < ApplicationController
   before_action :set_new_post, only: %i[create]
-  before_action :set_post, only: %i[show]
+  before_action :set_post, only: %i[show edit update accept reject destroy]
+
+  def index
+    # get all posts if current user is admin
+    # get only user's posts if current user is not admin
+    @posts = if current_user.admin?
+               Post.all.order(created_at: :desc).includes(%i[user tags])
+             else
+               current_user.posts.order(created_at: :desc).includes(%i[tags])
+             end
+
+    @posts = @posts.where(status: Post.statuses[params[:status]]) if params[:status]
+  end
 
   def show; end
 
   def new
     @post = Post.new
   end
 
+  def edit; end
+
   def create
     # remove blank tag value
     # convert from tag_id to tag object
     values = post_params
-    values['tags'].shift
-    values['tags'].map! { |tag| Tag.find(tag) }
+    values[:tags].shift
+    values[:tags].map! { |tag| Tag.find(tag) }
```

Comment:
You can reuse these code line with private method, define a private method that will return params with tags something like (check feature logic carefully again)
```
private

def post_params_with_tags
    // Insert code here
end
```